api/server.go

```diff
@@ -38,12 +65,123 @@ func (s *Server) walletBalanceHandler(w http.ResponseWriter, req *http.Request,
 	})
 }
 
+func (s *Server) walletAddressHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, WalletAddress{s.w.NextAddress()})
+}
+
+func (s *Server) walletAddressesHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, WalletAddresses{s.w.Addresses()})
+}
+
+func (s *Server) walletTransactionsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, Transactions{s.w.Transactions()})
+}
+
+func (s *Server) walletSignHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var txn types.Transaction
+	if err := json.NewDecoder(req.Body).Decode(&txn); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var amount types.Currency
+	for _, out := range txn.SiacoinOutputs {
+		amount = amount.Add(out.Value)
+	}
+
+	cm := s.s.ChainManager()
+	vc, err := cm.ValidationContext(cm.Tip())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if toSign, _, err := s.w.FundTransaction(&txn, amount, nil); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	} else if err := s.w.SignTransaction(vc, &txn, toSign); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	writeJSON(w, WalletTransaction{txn})
+}
+
+func (s *Server) walletBroadcastTransactionHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
```

Comment:
Broadcasting is a function of the txpool; nothing prevents you from broadcasting transactions that aren't relevant to your own wallet.